frontend/src/pages/projects/screens/spawner/storage/StorageClassSelect.tsx

```diff
@@ -0,0 +1,120 @@
+  HelperText,
+  HelperTextItem,
+} from '@patternfly/react-core';
+import { ExclamationTriangleIcon } from '@patternfly/react-icons';
+import React from 'react';
+import SimpleSelect, { SimpleSelectOption } from '~/components/SimpleSelect';
+import useStorageClasses from '~/concepts/k8s/useStorageClasses';
+import { getStorageClassConfig } from '~/pages/storageClasses/utils';
+
+type StorageClassSelectProps = {
+  storageClassName?: string;
+  setStorageClassName: (name: string) => void;
+  disableStorageClassSelect?: boolean;
+  menuAppendTo?: HTMLElement;
+};
+
+const StorageClassSelect: React.FC<StorageClassSelectProps> = ({
+  storageClassName,
+  setStorageClassName,
+  disableStorageClassSelect,
+  menuAppendTo,
+}) => {
+  const [storageClasses, storageClassesLoaded] = useStorageClasses();
+
+  const enabledStorageClasses = storageClasses
+    .filter((sc) => getStorageClassConfig(sc)?.isEnabled)
+    .toSorted((a, b) => {
+      const aConfig = getStorageClassConfig(a);
+      const bConfig = getStorageClassConfig(b);
+      if (aConfig?.isDefault) {
+        return -1;
+      }
+      if (bConfig?.isDefault) {
+        return 1;
+      }
+      return (aConfig?.displayName || a.metadata.name).localeCompare(
+        bConfig?.displayName || b.metadata.name,
+      );
+    });
+
+  const selectedStorageClass = storageClasses.find((sc) => sc.metadata.name === storageClassName);
+  const selectedStorageClassConfig = selectedStorageClass
+    ? getStorageClassConfig(selectedStorageClass)
+    : undefined;
+
+  const options: SimpleSelectOption[] = (
+    disableStorageClassSelect ? storageClasses : enabledStorageClasses
+  ).map((sc) => {
+    const config = getStorageClassConfig(sc);
+
+    return {
+      key: sc.metadata.name,
+      label: config?.displayName || sc.metadata.name,
+      description: config?.description,
+      isDisabled: !config?.isEnabled,
+      dropdownLabel: (
+        <Split>
+          <SplitItem>{config?.displayName || sc.metadata.name}</SplitItem>
+          <SplitItem isFilled />
+          <SplitItem>
+            {config?.isDefault && (
+              <Label isCompact color="green">
+                Default class
+              </Label>
+            )}
+          </SplitItem>
+        </Split>
+      ),
+    };
+  });
+
+  return (
+    <FormGroup label="Storage class" fieldId="storage-class" isRequired>
+      <SimpleSelect
+        dataTestId="storage-classes-selector"
+        id="storage-classes-selector"
+        isFullWidth
+        value={storageClassName}
+        options={options}
+        onChange={(selection) => {
+          setStorageClassName(selection);
+        }}
+        isDisabled={
+          disableStorageClassSelect || !storageClassesLoaded || storageClasses.length <= 1
+        }
+        placeholder="Select storage class"
+        popperProps={{ appendTo: menuAppendTo }}
+      />
+      <FormHelperText>
+        {selectedStorageClassConfig && !selectedStorageClassConfig.isEnabled ? (
```

Comment:
If a storage class config is not enabled, its considered deprecated? Are we sure this is the condition meant to use to indicate when a storage class is deprecated?

Reply:
not enabled is deprecated, thats what was in the mocks

Reply:
I don't consider this blocking since this is how the mocks describe it, but @xianli123 Is "Deprecated" the appropriate term to use here? Can't the user just turn around and enable the storage class from the storage classes table? I guess my question is why not just consider it "Disabled"?

Reply:
lets merge it now bc this pr blocks purva's, and we can circle back to it

Reply:
Hi @Gkrumbach07 @jpuzz0 Let me explain how to get the 'deprecated' class:

1. RHOAI admin enabled a class e.g. 'Class1'.
2. Users used the 'Class1' to create storage e.g. 'Storage1'.
3. RHOAI admin disabled 'Class1'.
4. Users will see 'Class1' is deprecated when checking 'Storage1'.

In a word, if the class is never enabled, it should not be a deprecated class, because users cannot select it. If the class has been enabled and used by some storage, after disabling the class, the class will be marked as 'Deprecated' in the storage that is using this newly disabled class.
Does it make sense?

Reply:
@xianli123 Thanks for explaining!

@Gkrumbach07 Are we checking to make sure the user created storage using the enabled class? Seems that is a prerequisite on top of the disabled state to mark the class as deprecated.